ui/src/components/search.tsx

```diff
@@ -6,17 +6,35 @@ import InputAdornment from "@mui/material/InputAdornment";
 import Stack from "@mui/material/Stack";
 import { TextField } from "mui-rff";
 import { Form } from "react-final-form";
+import { useSearchParams } from "react-router-dom";
 
 export type SearchProps = {
   placeholder?: string;
-  onSearch: (query: string) => void;
+  onSearch?: (query: string) => void;
+  initialValue?: string;
+  searchKey?: string;
 };
 
 export function Search(props: SearchProps) {
+  const setSearchParams = useSearchParams()[1];
```

Comment:
i had to look up this api to understand, this might be a more clear form:
const [_, setSearchParams]  = useSearchParams()

Reply:
Unfortunately it doesn't treat `_` specially and warns about it being unused.

Reply:
From this site it looks like you can do it, but just dropping the underscore and having the comma:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

Reply:
under "ignorign some returned values"

Reply:
That does work. Done.